src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpOnTypeFormattingPass.cs

```diff
@@ -106,7 +106,31 @@ public async override Task<FormattingResult> ExecuteAsync(FormattingContext cont
                 lineDelta++;
             }
 
-            var rangeToAdjust = new Range(rangeAfterFormatting.Start, new Position(rangeAfterFormatting.End.Line + lineDelta, 0));
+            // Now we know how many lines were affected by the cleanup and formatting, but we don't know where those lines are. For example, given:
+            //
+            // @if (true)
+            // {
+            //      }
+            // else
+            // {
+            // $$}
+            //
+            // When typing that close brace, the changes would fix the previous close brace, but the line delta would be 0, so
+            // we'd format line 6 and call it a day, even though the foratter made an edit on line 3.
```

Comment:
I think this comment does a great job of explaining the problem. I feel like it would also be helpful to explain how the added code is the proper solution (e.g. what `firstPosition` and `lastPosition` represent) so that people who are unfamiliar with the intricacies of the formatter can quickly decipher what's going on.